crates/ty_python_semantic/src/types/class.rs

```diff
@@ -4520,7 +4520,9 @@ impl KnownClass {
                 }
 
                 let bound_or_constraint = match (bound, constraints) {
-                    (Some(bound), None) => Some(TypeVarBoundOrConstraints::UpperBound(*bound)),
+                    (Some(bound), None) => Some(LazyTypeVarBoundOrConstraints::Eager(
+                        TypeVarBoundOrConstraints::UpperBound(*bound),
+                    )),
```

Comment:
optional nit: These would all be mildly shorter if you added `From` impls to create eager bounds, constraints, and defaults.

```suggestion
                    (Some(bound), None) => Some(TypeVarBoundOrConstraints::UpperBound(*bound).into()),
```